.github/workflows/hive-consensus-tests.yml

```diff
@@ -0,0 +1,4522 @@
+name: 'Hive consensus tests' 
+
+on:
+  push:
+    tags: ['*']
+  workflow_dispatch:
+    inputs:
+      parallelism:
+        description: 'Number of concurrently running tests in each job. With 1 or 2 timeout is likely. With 4 or more false-positive fails are likely. Recommended is 3 to avoid timeouts and reduce false-positives'
+        required: true
+        default: '3'
+        type: choice
+        options: ['1', '2', '3', '4', '8', '16']
+
+jobs:
+  test_1:
+    name: 1. Combined tests (e.g. stArgsZeroOneBalance)
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up parameters
+        run: |
+          echo "PARALLELISM=${{ github.event.inputs.parallelism || '3' }}" >> $GITHUB_ENV
+      - name: Check out Nethermind repository
+        uses: actions/checkout@v3
+        with:
+          path: nethermind
+      - name: Set up QEMU
```

Comment:
not sure if something can be cached: https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows

Reply:
Not sure too, I will left it as it is, might be potentially optimized in the future